### ISSUE TITLE
[Question] Get the inactive plugins

### ISSUE THREAD

**Issue by maxviewup** [thomasgriffin/TGM-Plugin-Activation#714]
Hello, i'm developing an manager plugin that integrates with TGMPA and others.

I need to get all the uninstalled/disabled plugins (or all the plugins status) to list them on an screen (similar to the TGMPA notice). Is there any way to get them? Maybe an TGMPA class method?

Also, if the first one is possible, i need to disable the TGMPA notice, then i will show my plugin notice that will have all the needed fixes, including TGMPA install.

**Comment by jrfnl** [comment]
Hi, I'm not entirely sure I understand your question correctly. Could you clarify after reading the below information ?

> I need to get all the uninstalled/disabled plugins (or all the plugins status) to list them on an screen (similar to the TGMPA notice). Is there any way to get them? Maybe an TGMPA class method?

TGMPA only lists the plugins which are passed to TGMPA as requirements by plugins/themes. There may be other plugins installed, but not active in the WP install. As that is outside of the scope of TGMPA, the library does not handle these.

TGMPA can only inform you about the plugins as registered with it.

> Also, if the first one is possible, i need to disable the TGMPA notice, then i will show my plugin notice that will have all the needed fixes, including TGMPA install.

Please realize that the list TGMPA generates is a combined list. I.e. if plugin A registers 3 required plugins, plugin B registers 1 recommended plugin and the active theme registers 2 required and 3 recommended plugins, the TGMPA admin notice will display just once with a message combining the required/recommended plugin info for both plugins and the theme.
For that matter, the TGMPA admin page will only be registered once and show the combined list as well.

So the TGMPA notice is not specific to the requirements of your plugin alone.

**Comment by maxviewup** [comment]
Hello, thanks for the answer.

> TGMPA can only inform you about the plugins as registered with it.

This is exactly what i need.

> Please realize that the list TGMPA generates is a combined list.

I know about this. The plugin i'm developing try to organize the WordPress UI and help user to setup it's site. it will give the user an list of TODOs explaining what he need to do and the link to fix the problem. 

That's why i need an list of all TGMPA required plugins not installed/activated (or if there is any to install/activate) and to remove the Admin notice (can be done with CSS/JS but using PHP would be great).

Basically, i will organize all plugins (that i'm supporting) reminders and needed actions from user (no need to do anything specifically for my plugin, just adding some hooks or functions will help).
